### PR TITLE
Support ONNX export for Streaming CTC Encoder

### PR DESCRIPTION
**Pull request by JinZr**

### CODE REVIEW

File: egs/librispeech/ASR/zipformer/export-onnx-streaming-ctc.py [hunk 1]
```diff
@@ -0,0 +1,570 @@
+#!/usr/bin/env python3
+#
+# Copyright 2023 Xiaomi Corporation (Author: Fangjun Kuang, Wei Kang, Zengrui Jin)
+# Copyright 2023 Danqing Fu (danqing.fu@gmail.com)
+
+"""
+This script exports a transducer model from PyTorch to ONNX.
```
Comment (csukuangfj) [inline egs/librispeech/ASR/zipformer/export-onnx-streaming-ctc.py:7]:
```suggestion
This script exports a CTC model from PyTorch to ONNX.
```

File: egs/librispeech/ASR/zipformer/export-onnx-streaming-ctc.py [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+  --left-context-frames 64 \
+  --use-ctc 1
+
+The --chunk-size in training is "16,32,64,-1", so we select one of them
+(excluding -1) during streaming export. The same applies to `--left-context`,
+whose value is "64,128,256,-1".
+
+It will generate the following file inside $repo/exp:
+
+  - ctc-streaming-encoder-epoch-99-avg-1-chunk-16-left-64.onnx
```
Comment (csukuangfj) [inline egs/librispeech/ASR/zipformer/export-onnx-streaming-ctc.py:44]:
```suggestion
  - ctc-epoch-99-avg-1-chunk-16-left-64.onnx
```

File: egs/librispeech/ASR/zipformer/export-onnx-streaming-ctc.py [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+
+See ./onnx_pretrained-streaming.py for how to use the exported ONNX models.
```
Comment (csukuangfj) [inline egs/librispeech/ASR/zipformer/export-onnx-streaming-ctc.py:46]:
```suggestion
See ./onnx_pretrained-streaming-ctc.py for how to use the exported ONNX models.
```

**Review by csukuangfj — APPROVED** [review]
Thanks!

**Comment by joazoa** [comment]
Thank you for the great work!
any chance you could make an export for the offline version as well ?

**Comment by JinZr** [comment]
Hi, thanks for asking.

You might want to check this script
https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/zipformer/export-onnx-ctc.py
for the non-streaming CTC model exporting. Please see if this one fits
your needs.

**Comment by joazoa** [comment]
@JinZr  is there already support for zipformer CTC offline for sherpa?

**Comment by JinZr** [comment]
yes, i think the support has been finalized for a while.

Best Regards
Jin


joazoa ***@***.***>于2023年12月23日 周六08:30写道：

**Pull request merged**